Forgot password for new login screen

bezumkin opened the pull request:
### What does it do?

- Removed system settings "manager_language" and "manager_lang_attribute".
- System settings "emailsubject" and "forgot_login_email" was moved to lexicons.
- Language of the manager is detected automatically and saved to session of user.
- User can change preferred language on login screen and it will work for "reset password emails" and the whole manager after login
- No more password in reset emails. User should confirm reset by link and specify new password manually.
- Removed ability to change password in manager and send it to user by email.
- Added Russian and Netherlands lexicons for new functions.
- Added pelago/emogrifier to improve HTML emails
- New default HTML template for manager emails
- Ability to notify user about creating an account for him
- Remove sending password from user create/update processors
- Added support of backgrounds to login screen

### Why is it needed?
To provide proper "forget password" workflow
and HTML emails
and other good things

### Related issue(s)/PR(s)
N/a

bezumkin committed:
- Removed system settings "manager_language" and "manager_lang_attribute".
- System settings "emailsubject" and "forgot_login_email" was moved to lexicons.
- Language of the manager is detected automatically and saved to session of user.
- User can change preferred language on login screen and it will work for "reset password emails" and the whole manager after login
- No more password in reset emails. User should confirm reset by link and specify new password manually.
- Removed ability to change password in manager and send it to user by email.
- Added Russian and Netherlands lexicons for new functions.

bezumkin committed:
- Added pelago/emogrifier to improve HTML emails
- New default HTML template for manager emails
- Ability to notify user about creating an account for him
- Remove sending password from user create/update processors
- Added support of backgrounds to login screen

Mark-H commented:
We'll need to make sure that the old (no longer used) settings are removed on upgrade, or that those setting values aren't used. Right now it's falling back to older versions of the email on an upgrade, which prevents the functionality from being used.

````emailsubject
forgot_login_email
manager_lang_attribute
manager_language
websignupemail_message
````

bezumkin committed:
- Fixed display of password in modal window when it changed on users update
- Removed lexicons string from english dictionary for deleted system settings
- Removed checking of deleted system settings in code

bezumkin committed:
- Fixed display of password in modal window when it changed on users update
- Removed lexicons string from english dictionary for deleted system settings
- Removed checking of deleted system settings in code

rthrash committed:
Updates to default English lexicon

bezumkin committed:
Merge branch '3x-forgot-password' of github.com:bezumkin/revolution into 3x-forgot-password

bezumkin committed:
Fixed name of system setting to check on registration notice

bezumkin committed:
Less margin before .lead

bezumkin committed:
- Remove "manager_language" setting on upgrade
- Final look improvements

bezumkin commented:
![screen shot 2018-03-16 at 13 23 02](https://user-images.githubusercontent.com/1257284/37520428-3dc85128-291d-11e8-812c-6bf228116a2b.png)
![screen shot 2018-03-16 at 13 23 08](https://user-images.githubusercontent.com/1257284/37520430-3ea01838-291d-11e8-9359-b129ae7b0d18.png)
![screen shot 2018-03-16 at 13 23 26](https://user-images.githubusercontent.com/1257284/37520432-3fc6939a-291d-11e8-8c76-4813f31eee7d.png)

Mark-H commented:
🎉
